lib/tools/node-modules/node-modules-builder.ts

```diff
@@ -133,27 +133,165 @@ export class NodeModulesBuilder implements INodeModulesBuilder {
 				// Force copying if the destination doesn't exist.
 				lastModifiedTime = null;
 			}
-			let nodeModules = this.getChangedNodeModules(absoluteOutputPath, platform, lastModifiedTime).wait();
 
-			const resolver = new NpmDependencyResolver(this.$projectData.projectDir);
-			const resolvedDependencies = resolver.resolveDependencies(_.keys(nodeModules), platform);
+			let productionDependencies = this.getProductionDependencies(this.$projectData.projectDir);
+
+			// console.log(productionDependencies);
+
+			// TODO: Pip3r4o - result is not used currently
+			// let nodeModules = this.getChangedNodeModules(absoluteOutputPath, platform, lastModifiedTime).wait();
 
 			if (!this.$options.bundle) {
 				const tnsModulesCopy = this.$injector.resolve(TnsModulesCopy, {
 					outputRoot: absoluteOutputPath
 				});
-				tnsModulesCopy.copyModules(resolvedDependencies, platform);
+				tnsModulesCopy.copyModules(productionDependencies, platform);
 			} else {
 				this.cleanNodeModules(absoluteOutputPath, platform);
 			}
 
 			const npmPluginPrepare = this.$injector.resolve(NpmPluginPrepare, {});
-			npmPluginPrepare.preparePlugins(resolvedDependencies, platform);
+			npmPluginPrepare.preparePlugins(productionDependencies, platform);
 		}).future<void>()();
 	}
 
+	public getProductionDependencies(projectPath: string) {
+		var deps: any = [];
+		var seen: any = {};
+
+		var pJson = path.join(projectPath, "package.json");
+		var nodeModulesDir = path.join(projectPath, "node_modules");
+
+		var content = require(pJson);
+
+		Object.keys(content.dependencies).forEach((key) => {
+			var depth = 0;
+			var directory = path.join(nodeModulesDir, key);
+
+			// find and traverse child with name `key`, parent's directory -> dep.directory
+			traverseChild(key, directory, depth);
+		});
+
+		return filterUniqueDirectories(deps);
+
+		function traverseChild(name: string, currentModulePath: string, depth: number) {
```

Comment:
Those inner functions are very confusing and unreadable. Consider moving them to methods of this class or an entirely new class.